pyEpiabm/pyEpiabm/core/_compartment_counter.py

```diff
@@ -47,9 +53,17 @@ def report(self, old_status: InfectionStatus,
         :param new_status: Person's new infection state
         :type new_status: InfectionStatus
         """
+        if self._compartments[old_status] == 0:
+            raise ValueError("No people of this status in this cell.")
         self._compartments[old_status] -= 1
         self._compartments[new_status] += 1
 
+    def report_new_person(self) -> None:
```

Comment:
Is this function just here for setting up the population? We're not actually adding new people mid-simulation?
The change you made to initialize is perfect, and in the C++ I just have it called by the simulation class right before starting the iterations. I don't have any problem doing it this way instead, but should maybe name it something else or at least document if it's just for initialization

Reply:
Think I replied on teams:

I kept add new person separate primarily for the testing where you're changing the population around a lot its needed. Can definitely see use in a full simulation though, maybe if infected people are coming into the country from abroad. Hopefully none of the compartment counters should be called explicitly now, it's all wrapped up in either add_person to microcell of in the population factory. So will document that